File: 21_10_7/point.py
import math
class point(object):
    def __init__(self,x=0,y=0):
        self.x = x
        self.y = y
    def __str__(self):
        return f"{self.x},{self.y}"
    def __eq__(self, other):
        if type(other) != point:
            return False
        return self.x == other.x and self.y == other.y

    def distance_to(self, other):
        return math.hypot(self.x - other.x, self.y - other.y)
    def reflect_across_x(self):
        return(self.x,-self.y)

p = point(3,4)
s = point(3,4)
q = point(0,0)
print(f"P == S: {p == s}")
print(f"Distance to points: {p.distance_to(q)}")
#print(p)
print(f"Reflect across X: {p.reflect_across_x()}")

File: 21_10_7/point.java
public class point {
    private final int x;
    private final int y; //final means "cannot be reassigned"

    public point(int x,int y){
        this.x = x;
        this.y = y;
    }
    /* 
     * method name overloading
     * several methods can have the same name if their
     * sigs are different
     * 
     * You should only use this for closeley relared methods
     */
    public point(){ //sig is []
        //x = 0;
        //y = 0;
        this(0,0); //calls sibling constructor
    }
    public int getX(){
        return x;
    }
    public int getY(){
        return y;
    }
    
    @Override
    public String toString(){
        return String.format("%s,%s", x, y);
    }
    @Override
    public boolean equals(Object o){
        if(this == o){
            return true;
        }
        if(!(o instanceof point)){
            return false;
        }
        //o must be a point
        point that = (point) o;
        return x == that.x && that.y == y;
    }

    public double distanceTo(point other){
        return Math.hypot(x-other.x, y - other.y);
    }
    
    public point reflectAcrossX(){
        return new point(x, -y);
    }public point reflectAcrossY(){
        return new point(-x, y);
    }
    public point reflectAcrossXequalsY(){
        return new point(x, y);
    }
    
    public static void main(String[] args){
        point p = new point(3,4);
        point q = new point(0,0);
        point s = new point(3,4);
        System.out.print("P == S: ");
        System.out.println(p == s);
        //System.out.println(p.x);
        //System.out.println(p.y);
        System.out.print("Distance to points: ");
        System.out.println(p.distanceTo(q));
        //System.out.println(p);
        System.out.print("Reflect across X: ");
        System.out.println(p.reflectAcrossX());
    }
}